sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -158,6 +158,21 @@ trait CatalystScan extends BaseRelation {
 }
 
 @DeveloperApi
+/**
+ * ::DeveloperApi::
+ * A BaseRelation that can be used to insert data into it through the insert method.
+ * If overwrite in insert method is true, the old data in the relation should be overwritten with
+ * the new data. If overwrite in insert method is false, the new data should be appended.
+ *
+ * InsertableRelation has the following three assumptions.
+ * 1. It assumes that the data (Rows in the DataFrame) provided to the insert method
+ * exactly matches the ordinal of fields in the schema of the BaseRelation.
+ * 2. It assumes that the schema of this relation will not be changed.
+ * Even if the insert method updates the schema, the new schema cannot be used.
```

Comment:
Would be good to mention the schema evolving cases in JSON and Parquet here. I guess it's not super intuitive for most people because stable schema is a basic assumption for traditional RDBMS.